### PR TITLE
SEAR-146 query types per field for facility search

### PR DESCRIPTION
**Pull request by ratneshraval**
#### Which User story does this relate to (link)?
- https://osi-cwds.atlassian.net/browse/SEAR-146

#### Brief feature description
- Facility search query now supports query_type per field

### CODE REVIEW

**Comment by ratneshraval** [comment]
jenkins, retest this please

**Comment by Nasir2712** [comment]
if possible can you take a look at these ruby lint errors?

File: spec/classes/elastic/query_builder_spec.rb [hunk 1]
```diff
@@ -72,123 +68,83 @@
...
         from:"0",
         size:"5",
         sort: []
       }
-      query_array = [{"county.value"=>"Los Angeles", "type.value"=>"Resource Family Home", "id"=>"123124", "name"=>"home"}]
+
+      query_array = [
+        {"county.value": {query_type: 'match', value: "Los Angeles"},
+         "type.value":{query_type: 'term', value: "Resource Family Home"},
+         "id": {query_type: 'match_phrase', value: "123124"},
```
Comment (Nasir2712) [inline spec/classes/elastic/query_builder_spec.rb]: should be license number instead of id?

Comment (ratneshraval) [reply spec/classes/elastic/query_builder_spec.rb]: Yes, intentionally kept it as `id`.

File: spec/classes/query_preprocessor_spec.rb [hunk 2]
```diff
@@ -3,40 +3,80 @@
...
-                          {:county=>'02', :type=>'2', :fac_nbr=>'90'},
-                          {:county=>'02', :type=>'4', :fac_nbr=>'80'},
-                          {:county=>'02', :type=>'4', :fac_nbr=>'90'}]
-        params = {'county': ['01', '02'], 'type': ['2', '4'], 'fac_nbr': ['80', '90']}
-        output = QueryPreprocessor.params_to_query_hash(params)
-
-        expect(output).to eq(expected_output)
-      end
+  describe 'params_to_query_with_types' do
+    it 'converts input params to hash with all combinations' do
```
Comment (Nasir2712) [inline spec/classes/query_preprocessor_spec.rb]: awesome test case 👍

**Comment by Nasir2712** [comment]
A new PR will be created to address the ```address``` query updates implemented by the backend.

**Pull request merged**